SETUP/upgrade/13/20191228_alter_page_events.php

```diff
@@ -0,0 +1,25 @@
+<?php
+$relPath='../../../pinc/';
+include_once($relPath.'base.inc');
+
+header('Content-type: text/plain');
+
+// ------------------------------------------------------------
+
+echo "Dropping redundant keys on page_events...\n";
+$sql = "
+    ALTER TABLE page_events
+        DROP INDEX projectid,
```

Comment:
I'm pretty sure the 'projectid' key is being used. Look at `archive_ancillary_data_for_project`, which (via `move_project_rows_to_archive_db`) archives rows for a given project from `page_events`. `EXPLAIN` says that it uses the `projectid` index.

Moreover, when a proofer reports something odd relating to pages checked out/saved/returned/whatever, we typically do ad hoc queries like `select * from page_events where project='...' and ...`, which will probably use the `projectid` index.

Side note: the `projectid` index is an index on (`projectid`,`image`,`round_id`), but it probably doesn't need to be that granular. For a given project, there's probably only about a couple thousand page events, and if a query is only interested in a subset of them, I'm guessing MySQL can filter them pretty quickly even without a more granular index. So we could maybe save space by dropping `image` and `round_id` from the index.

Reply:
I missed that (yay for code reviews!). MySQL seems happy to use either the `projectid` index or the `projectid_username` index for that query for whichever exists. Perhaps we keep these two and remove the other two?
```
  KEY `username` (`username`,`round_id`),
  KEY `projectid_username` (`projectid`,`username`),
```

Reply:
So removing:
```
KEY `projectid` (`projectid`,`image`,`round_id`),
KEY `username_projectid_round_time` (`username`,`projectid`,`round_id`,`timestamp`)
```
That seems like a good plan to me, but I've been wrong about indexes before. Is there a way we can test their removal on prod such that if it doesn't work, we can reinstate them quickly?

Reply:
(I'm assuming that rebuilding them from scratch would not be quick.)

Reply:
Dropping the index should be very fast as a simple metadata operation. Adding it back would be slow, but the table would be available the entire time because that's an online DDL operation (see https://dev.mysql.com/doc/refman/5.7/en/innodb-online-ddl-operations.html). I don't know how we can test it on prod without doing the changes and I also don't know how we could do a backup such that we could quickly fall back.